SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -99,42 +100,172 @@ namespace {
+  typedef std::map<uint32_t, std::vector<std::pair<float, float>>> IdHit_Map;
+  template <typename GEOM>
+  void loadSimHitAccumulator_forPreMix(hgc::HGCSimHitDataAccumulator& simData,
+                                       hgc::HGCPUSimHitDataAccumulator& PUSimData,
+                                       const GEOM* geom,
+                                       IdHit_Map& hitRefs_bx0,
+                                       const PreMixHGCSimAccumulator& simAccumulator,
+                                       const float minCharge,
+                                       const float maxCharge,
+                                       bool setIfZero,
+                                       const std::array<float, 3>& tdcForToAOnset,
+                                       const bool& minbiasFlag,
+                                       std::map<uint32_t, bool>& hitOrder_monitor) {
+    std::cout << hitOrder_monitor.size() << std::endl;
+    const float minPackChargeLog = minCharge > 0.f ? std::log(minCharge) : -2;
+    const float maxPackChargeLog = std::log(maxCharge);
+    constexpr uint16_t base = 1 << PreMixHGCSimAccumulator::SimHitCollection::sampleOffset;
+    for (const auto& detIdIndexHitInfo : simAccumulator) {
+      unsigned int detId = detIdIndexHitInfo.detId();
+
+      auto PUsimIt = PUSimData.emplace(detId, HGCCellHitInfo()).first;
+      auto simIt = simData.emplace(detId, HGCCellInfo()).first;
+      auto& puhit_info = PUsimIt->second.PUhit_info;
+      size_t nhits = detIdIndexHitInfo.nhits();
+
+      hitOrder_monitor[detId] = false;
+      if (nhits > 0) {
+        unsigned short iSample = detIdIndexHitInfo.sampleIndex();
+        std::vector<unsigned short> unsigned_charge_array = detIdIndexHitInfo.chargeArray();
+        std::vector<unsigned short> unsigned_time_array = detIdIndexHitInfo.timeArray();
+        std::vector<float> Acc_charge_array = puhit_info[0][iSample];
+        std::vector<float> timing_hit_array = puhit_info[1][iSample];
+
+        float p_charge, p_time;
+        unsigned short unsigned_charge, unsigned_time;
+
+        for (size_t ihit = 0; ihit < nhits; ++ihit) {
+          unsigned_charge = (unsigned_charge_array[ihit] & HGCDigitizer::dataMask);
+          unsigned_time = (unsigned_time_array[ihit] & HGCDigitizer::dataMask);
+          p_time = logintpack::unpack16log(unsigned_time, minPackChargeLog, maxPackChargeLog, base);
+          p_charge = logintpack::unpack16log(unsigned_charge, minPackChargeLog, maxPackChargeLog, base);
+          const std::pair<float, float> charge_time_pair = std::make_pair(p_charge, p_time);
+          (simIt->second).hit_info[0][iSample] += p_charge;
+          if (iSample == 9) {
+            if (hitRefs_bx0[detId].empty()) {
+              hitRefs_bx0[detId].emplace_back(charge_time_pair);
+            } else {
+              if (p_time < hitRefs_bx0[detId].back().second) {
+                itr findPos = std::upper_bound(hitRefs_bx0[detId].begin(),
```

Comment:
Here and below, how about using `auto`
```suggestion
                auto findPos = std::upper_bound(hitRefs_bx0[detId].begin(),
```
instead of doing a `typedef` somewhere way up for a very short type name?